### ISSUE TITLE
"approximation" argument of Estimator is poorly named

### ISSUE THREAD

**Issue by kevinsung** [Qiskit/qiskit-aer#1824]
### What is the expected behavior?

It's somewhat confusing that setting `approximation=True` is necessary to get the _exact_ expectation value. For example, consider this sentence from the [documentation](https://qiskit.org/ecosystem/aer/stubs/qiskit_aer.primitives.Estimator.html):
> If None and approximation is True, it calculates the exact expectation values.

I suggest renaming the argument to `simulate_sampling_error` or just `simulate_error`.

**Comment by athanzxyt** [comment]
Hi @hhorii this PR is ready for review.

**Comment by ikkoham** [comment]
Yes. This requires some `DeprecationWarning`. I'll redesign overall with `options` (with flexible `dataclass`) align with `qiskit-ibm-runtime`.
